src/shared/Microsoft.Git.CredentialManager/ConfigurationService.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Git.CredentialManager
+{
+    /// <summary>
+    /// Component that requires Git or environment configuration to work.
+    /// </summary>
+    public interface IConfigurableComponent
+    {
+        /// <summary>
+        /// Name of the component that requires configuration.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// Configure the environment and Git to work with this hosting provider.
+        /// </summary>
+        /// <param name="system">Configure the component for all users on the system.</param>
+        /// <param name="environment">Environment variables.</param>
+        /// <param name="configuration">User-level Git configuration.</param>
+        Task ConfigureAsync(bool system, IEnvironment environment, IGitConfiguration configuration);
```

Comment:
How about an enum rather than the bool to make this a bit simpler to grok? We already have `GitConfigurationLevel`.

Reply:
Added and plumbed through a new `ConfigurationTarget` enumeration!